src/test/scala/com/vngrs/etl/CombineSpec.scala

```diff
@@ -0,0 +1,62 @@
+package com.vngrs.etl
+
+/**
+  * Combiner Specifications
+  */
+class CombineSpec extends BaseSparkSpec {
+
+  "A functionless Combiner" should "return given Pipeline with an Empty RDD as it is" in {
+    val rdd = emptyRDD[String]
+    val pipeline = new Pipeline(rdd)
+
+    val combiner = new Combine[String, String] {
+      override def apply(pipeline2: Pipeline[String]): Pipeline[String] = pipeline2
+    }
+
+    combiner(pipeline) should equal (pipeline)
+  }
+
+  it should "return given Pipeline with any non empty RDD without changing data in it" in {
+    val data = List("a", "b", "c")
+    val rdd = parallelize(data)
+    val pipeline = new Pipeline(rdd)
+
+    val combiner = new Combine[String, String] {
+      override def apply(pipeline2: Pipeline[String]): Pipeline[String] = pipeline2
+    }
+
+    combiner(pipeline).testCollect should contain theSameElementsAs data
+  }
+
+  "A counter Combiner" should "accept a Pipeline with an empty RDD" in {
+    val pipeline1 = new Pipeline(emptyRDD[Int])
+    val pipeline2 = new Pipeline(emptyRDD[Int])
+
+    val combiner = Combine[Int, Int, Long]((rdd1, rdd2) => parallelize(List(rdd1.count(), rdd2.count())), pipeline1)
+
+    val combinedPipeline = combiner(pipeline2)
```

Comment:
Let's use our DSL on tests too. `pipeline1.combine(combiner)` will be much more clear.